src/earcut.js

```diff
@@ -344,7 +344,8 @@ function findHoleBridge(hole, outerNode) {
 
             tan = Math.abs(hy - p.y) / (hx - p.x); // tangential
 
-            if ((tan < tanMin || (tan === tanMin && p.x > m.x)) && locallyInside(p, hole)) {
+            if ((tan < tanMin || (tan === tanMin && (p.x > m.x || area(m, m.next, p.next) > 0))) &&
```

Comment:
I think the issue is that if the tangent is vertical (tanMin == Inf), and there are multiple coincident possible bride points, there is an issue, because it chooses the first one, instead of the closest. Instead of `p.x > m.x`, it should probably check `p.x*p.x+p.y*p.y > m.x*m.x+m.y*m.y`. I'll look into it later

Reply:
No, it's not that. See this picture of when it fails:

![image](https://user-images.githubusercontent.com/25395/65038833-8187fb00-d959-11e9-8d52-59a66a967cc5.png)

For the last hole, there are multiple nodes in the same spot, with edges going into different directions. We're picking the node that belonged to the second hole (producing invalid connection), rather than picking the node that has the diagonal edge. So, to fix that, for multiple candidates with the same exact coordinates, we want to pick one with an edge closest to the hole point when going counter-clockwise. So I think the fix direction is right, I'm just not confident it covers all similar cases. But better than nothing.